crates/telio-firewall/src/firewall.rs

```diff
@@ -546,7 +540,7 @@ impl StatefullFirewall {
         let flags = tcp_packet.get_flags();
         let mut tcp_cache = unwrap_lock_or_return!(self.tcp.lock(), false);
 
-        if let Some(connection_info) = tcp_cache.get(&key) {
+        if let Some(connection_info) = tcp_cache.peek(&key) {
```

Comment:
Please add a test that would fail before this change. Maybe something like sending non `ACK` packets when `!connection_info.tx_alive
                && !connection_info.rx_alive
                && !connection_info.conn_remote_initiated` and verify that such packets don't extend the entry lifetime.

Reply:
added test `firewall_tcp_connection_timeout_is_not_updated_after_fin`